File: data_structures_and_algorithms_on_graphs/adjacency matrix/data_structures_and_algorithms_on_graphs_0.1.py
inFile = open('input.txt', 'r')
outFile = open('output.txt', 'w')
lines = inFile.readlines()
inFile.close()
n_m = list(map(int, (lines[0]).split()))
a = []

for i in range(n_m[0]):
    b = []
    for j in range(n_m[0]):
        x = 0
        b.append(x)
    a.append(b)
    
for edge in lines[1:]:
    edge = list(map(int, edge.split()))
    a[edge[0] - 1][edge[1] - 1] = 1
    a[edge[1] - 1][edge[0] - 1] = 1
    print(a)

for i in range(n_m[0]):
    for j in range(n_m[0]):
        print(a[i][j], file=outFile, end=' ')
    print('\n', file=outFile, end='')
outFile.close()

File: data_structures_and_algorithms_on_graphs/adjacency matrix/data_structures_and_algorithms_on_graphs_0.1.cpp
#include <iostream>
#include <string>
#include <fstream>

using namespace std;

int matr[110][110];

int main() {
    ifstream in("input.txt");
    int n, m, a, b;
    in >> n >> m;
    for (int i = 0; i < m; i++) {
        in >> a >> b;
        matr[a - 1][b - 1] = 1;
        matr[b - 1][a - 1] = 1;
    }
    in.close();
    ofstream out("output.txt");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            out << matr[i][j] << " ";
        }
        out << endl;
    }
    out.close();
    return 0;
}